### ISSUE TITLE
Add an option or command to dump `dvc run` by a stage name

### ISSUE THREAD

**Issue by shcheklein** [iterative/dvc#4002]
I suppose we have enough information to construct a command that was used to create the stage from a dvc.yaml file?

It would help a lot in editing the file, scripting, etc.

**Comment by efiop** [comment]
@shcheklein The usecase is not really clear. Could you elaborate?

**Comment by shcheklein** [comment]
Yep, sure. Let's say I have a project with `dvc.yaml`:

```yaml
stages:
  prepare:
    cmd: python src/prepare.py data/data.xml
    deps:
    - data/data.xml
    - src/prepare.py
    outs:
    - data/prepared
```

and I don't a script/history to get the initial `dvc run` command that was used to create this file.

Now, imagine I decided to introduce params and/or metrics. I changed the `src/prepare.py` to. include:

```python
import yaml

params = yaml.safe_load(open('params.yaml'))['prepare']

# Test data set split ratio
split = params['split']
random.seed(params['seed'])
...
```

it means that I need to run `dvc run` again or manually update the `dvc.yaml` to get it to something like:

```yaml
stages:
 prepare:
    cmd: python src/prepare.py data/data.xml
    deps:
    - data/data.xml
    - src/prepare.py
    params:
    - prepare.seed
    - prepare.split
    outs:
    - data/prepared
```

In this simplified example, probably not that much difference, but if I want to do a bit more sophisticated changes, it would be great to have something like:

`dvc repro --dump prepare` that outputs:

`dvc run --overwrite -d src/prepare.py -d data/data.xml -o data/prepared python src/prepare.py data/data.xml`

so that one can quickly fix, adding `-n prepare -p prepare.split,prepare.seed` and run it again.

**Comment by efiop** [comment]
But `dvc.yaml` was meant to be adjusted by hand, and it is far easier than running a command to generate the `dvc run` command and then adjusting that (possibly big) command to run it again.

**Comment by shcheklein** [comment]
@efiop it was definitely not easier for me when I was adding params for the first time :) I was faster for me to get the previous command and edit it.

**Comment by efiop** [comment]
@shcheklein But you've got it from the shell history, which is easy. Generating it back using a special dvc command and then copypasting it to edit and then run it feels a lot harder.

**Comment by skshetry** [comment]
@shcheklein, you can copy-paste-edit from `dvc.yaml` and then `dvc repro`.  I'd love to not have a command for these cases, we should direct to changing `dvc.yaml` by hand more.

**Comment by shcheklein** [comment]
> But you've got it from the shell history,

I didn't have it from the shell history (neither do other users always have it). (I had to jump to the GH online repo and extract it from the gen script - and still it was easier for me, at least first time - since I just didn't know what is the exact expected syntax and what sections should I update.)

> you can copy-paste-edit from dvc.yaml and then dvc repro

not sure I got it, to be honest

>  I'd love to not have a command for these cases, we should direct to changing dvc.yaml by hand more

yep, me too. But, since I just had to do this, it feels painful. In the ideal world all users will know our syntax in advance and won't use the command at all. In the real world learning a new DSL from scratch is painful. Thus we provide `dvc run`, right? (should we remove it as well? and ask everyone to write stages manually?)

The only real concern that I have is that even today probably we don't have exact mapping from `dvc.yaml` to `dvc run`. Stage stage syntax is already too complicated to be handled in via CLI? Or it will be soon. And this is a real deal breaker for this probably.

I will try to improve params docs after I'm done with the get started.

Closing this for now.
